### DISCUSSION TITLE
Python Library for Birdbrain Techologies

### DISCUSSION THREAD

**Discussion by fmorton** [mu-editor/mu#2150] [General]
We are consistently using Birdbrain Technologies products at our makerspace now using the mu editor. It is absolutely fantastic in this application.

Birdbrain distributes a single python class file to control their robots. I am guessing licensing issues would not be problem and they would be adaptive and supportive of this, but thought I would ask here first.

The hope would be to place their library file in:

/Applications/Mu Editor.app/Contents/Resources/Python/lib/python3.8

The library that needs to be included is their BirdBrain.py file found at:

https://github.com/BirdBrainTechnologies/BirdBrain-Python-Library

I would be happy to be the responsible person to keep this up-to-date as needed, though updates are infrequent.

This would make all the BirdBrain products (hummingbird and finch) ready out of the box with mu.

Thanks.

**Comment by carlosperate** [comment]
Having the library inside the packaged Python environment will not make it accessible to user code, as a virtual environment is created for the user and packages are installed there.

Is there any reason Birdbrain Technologies doesn't publish the package in PyPI? That way it can be installed via the Mu settings.

**Reply by fmorton** [reply]
Followed your advice and put the package in pypi.org.

https://www.youtube.com/watch?v=kOHHkjRb3p8
